code/game/turfs/simulated/floor/plating/asteroid.dm

```diff
@@ -136,6 +104,7 @@
 
 /turf/open/floor/plating/asteroid/basalt/Initialize()
 	. = ..()
+	AddComponent(/datum/component/archaeology/basalt, 100)
```

Comment:
I may be dumb but this won't add both /datum/component/archaeology/ [from inher] AND /basalt will it?

Reply:
Separate out the base asteroid path and basalt path. Generally speaking if there's something in a parent type you don't want one of its children to do, that implies you're misusing inheritance.

Reply:
I may be misreading but it seems like a lot of copy/paste just to have one thing [that is which component to add] differ. I could add a check to asteroid initialize to see if there's already an existing archaeological datum but that seems odd.